### PR TITLE
Snapshot bump

### PR DESCRIPTION
**Pull request by roypat**
## Changes

Cherry-picks the snapshot version bump from the 1.3 release branch, as well as introduces a regression test (thanks to pablob@ for the code!) that ensure we do not forget to bump snapshot version prior to a release again.

## Reason

We forgot to bump snapshot version prior to releasing 1.3.0

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/functional/test_snapshot_version.py [hunk 1]
```diff
@@ -116,6 +119,47 @@ def test_create_invalid_version(bin_cloner_path):
         assert False, "Negative test failed"
 
 
+def test_snapshot_current_version(bin_cloner_path):
+    """Tests taking a snapshot at the version specified in Cargo.toml
+
+    Check that it is possible to take a snapshot at the version of the upcoming
+    release (during the release process this ensures that if we release version
+    x.y, then taking a snapshot at version x.y works - something we'd otherwise
+    only be able to test once the x.y binariy has been uploaded to S3, at which
```
Comment (pb8o) [inline tests/integration_tests/functional/test_snapshot_version.py:128]: typo binary

**Review by pb8o — APPROVED** [review]

File: tests/integration_tests/functional/test_snapshot_version.py (hunk 1 above)
Comment (roypat) [reply tests/integration_tests/functional/test_snapshot_version.py]: fixed :D

**Review by zulinx86 — APPROVED** [review]
LGTM.

> thanks to pablob@ for the code!

If pablob@ contributes the script, how about adding him as "Co-authored-by"?

**Review by luminitavoicu — DISMISSED** [review]

**Comment by roypat** [comment]
> LGTM.
> 
> > thanks to pablob@ for the code!
> 
> If pablob@ contributes the script, how about adding him as "Co-authored-by"?

ah, yes, forgot those existed. Thanks for the call out!

**Review by luminitavoicu — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Review by zulinx86 — APPROVED** [review]

**Pull request merged**